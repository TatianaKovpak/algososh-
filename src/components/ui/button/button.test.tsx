import { Button } from "./button";
import renderer from 'react-test-renderer'
import { render, fireEvent } from '@testing-library/react';


describe('Тестирование кнопки', () => {
    it('С текстом', () => {
        const button = renderer.create(<Button text = 'Текст' type='button'/>).toJSON()
        expect(button).toMatchSnapshot()
    })

    it('Без текста', () => {
        const button = renderer.create(<Button type='button'/>).toJSON()
        expect(button).toMatchSnapshot()
    })

    it('Неактивная кнопка', () => {
        const button = renderer.create(<Button type='button' disabled={true}/>).toJSON()
        expect(button).toMatchSnapshot()
    })

    it('С лоадером', () => {
        const button = renderer.create(<Button type="button" isLoader={true}/>).toJSON()
        expect(button).toMatchSnapshot()
    })

    it('Вызов колбэка при клике', () => {
      const onClickMock = jest.fn()
 
      const { getByText } = render(<Button onClick={onClickMock} type='button' text='Текст' />);
      const button = getByText('Текст');
      fireEvent.click(button);

      expect(onClickMock).toHaveBeenCalled();
    })
})
