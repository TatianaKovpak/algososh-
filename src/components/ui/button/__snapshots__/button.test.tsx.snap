// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование кнопки Без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование кнопки Неактивная кнопка 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование кнопки С лоадером 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование кнопки С текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст
  </p>
</button>
`;
